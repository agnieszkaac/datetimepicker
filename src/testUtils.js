export const setDate = (date) => {
  Date.now = jest.fn(() => Date.parse(date));
};

export const setToday = () => {
  const today = "2020-01-01";
  Date.now = jest.fn(() => Date.parse(today));
};
