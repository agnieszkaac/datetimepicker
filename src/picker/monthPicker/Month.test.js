import React from "react";
import moment from "moment";
import { fireEvent, render } from "@testing-library/react";

import { Month } from "./";
import { setToday } from "../../testUtils";

describe("<Month /> should", () => {
  setToday();

  //default props
  const date = moment();

  let value = null;
  const onClick = jest.fn((e) => (value = e.target.value));
  const props = {
    date,
    selected: false,
    onClick,
  };
  const displayValue = "Jan";
  const expectedValue = date.toISOString();

  it("match snapshot", () => {
    const { asFragment } = render(<Month {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("have correct display value for default date", () => {
    const { getByTestId } = render(<Month {...props} />);
    const month = getByTestId("month");

    expect(month).toHaveTextContent(displayValue);
  });
  it("have selected class when selected", () => {
    const { getByTestId, rerender } = render(<Month {...props} />);
    const day = getByTestId("month");

    expect(day).not.toHaveClass("selected");

    rerender(<Month {...props} selected={true} />);
    expect(day).toHaveClass("selected");
  });
  it("fire onClick with proper value when clicked", () => {
    const { getByTestId } = render(<Month {...props} />);
    const month = getByTestId("month");

    expect(month).toHaveValue(expectedValue);

    fireEvent.click(month);

    expect(onClick).toHaveBeenCalledTimes(1);
    expect(value).toBe(expectedValue);
  });
});
