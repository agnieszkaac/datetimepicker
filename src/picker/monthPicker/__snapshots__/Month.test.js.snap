// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Month /> should match snapshot 1`] = `
<DocumentFragment>
  <button
    class="month is-today"
    data-testid="month"
    type="button"
    value="2020-01-01T00:00:00.000Z"
  >
    Jan
  </button>
</DocumentFragment>
`;
